runelite-client/src/main/java/net/runelite/client/plugins/fairyring/FairyRingPlugin.java

```diff
@@ -316,10 +354,13 @@ else if (!Strings.isNullOrEmpty(w.getText()))
 				}
 			}
 
+			String finalFilter = filter;
 			boolean hidden = !(filter.isEmpty()
 				|| Text.removeTags(c.getDescription().getText()).toLowerCase().contains(filter)
 				|| code.toLowerCase().contains(filter)
-				|| tags != null && tags.contains(filter));
+				|| tags != null && tags.contains(filter)
+				|| (enableTagSearching && tags != null && getTags(code).stream().anyMatch(s -> s.contains(finalFilter))
```

Comment:
I had though that a set contains would match on exact tags being met rather than a partial matching, so I am checking the set to see if any of the Strings within the Collection of Strings match partially the way I knew how.